src/ai/defaultai.cc

```diff
@@ -1375,7 +1375,7 @@ void DefaultAI::update_all_buildable_fields(const Time& gametime) {
 	// Stage #2: get rid of invalid files / and rotate the deque
 	uint16_t min_fields_rotated =
 	   buildable_fields.size() / 15;  // rotating at least this number of items
-	while (invalidated_bf_count || min_fields_rotated--) {
+	while ((invalidated_bf_count != 0u) || ((min_fields_rotated--) != 0u)) {
```

Comment:
I am unsure whether the min_fields_rotated value is still evaluated against the previous state before dekrement or if it is now evaluated against the dekremented values due to the brackets around it.

Reply:
This is guaranteed by the ISO standard: The expression `min_fields_rotated--` evaluates to the value of `min_fields_rotated` **before** the decrement. `--min_fields_rotated` would evaluate to the value **after** the decrement.

The extra parentheses are just for readability; `expr` and `(expr)` always evaluate to the same value.